### ISSUE TITLE
Error when creating "azurerm_storage_queue", since v3.96.0

### ISSUE THREAD

**Issue by oscaralvreyes** [hashicorp/terraform-provider-azurerm#25291]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.5

### AzureRM Provider Version

3.96.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_queue

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "cmgt-st" {
  name                     = var.st-name
  resource_group_name      = var.rg-name
  location                 = var.st-location
  account_tier             = "Standard"
  account_replication_type = "LRS"

  blob_properties {
    cors_rule {
      allowed_headers = ["*"]
      allowed_methods = ["GET","HEAD"]
      allowed_origins = ["*"]
      exposed_headers = ["*"]
      max_age_in_seconds = 0
    }
  }
}

resource "azurerm_storage_queue" "st-cm-documents-created" {
  name                 = var.st-cm-documents-created
  storage_account_name = azurerm_storage_account.cmgt-st.name
}

resource "azurerm_storage_queue" "st-cm-connection-accepted" {
  name                 = var.st-cm-connection-accepted
  storage_account_name = azurerm_storage_account.cmgt-st.name
}

resource "azurerm_storage_queue" "st-cm-connection-rejected" {
  name                 = var.st-cm-connection-rejected
  storage_account_name = azurerm_storage_account.cmgt-st.name
}

resource "azurerm_storage_queue" "st-cm-user-updated" {
  name                 = var.st-cm-user-updated
  storage_account_name = azurerm_storage_account.cmgt-st.name
}

resource "azurerm_storage_queue" "st-cm-scheduled-task-triggered" {
  name                 = var.st-cm-scheduled-task-triggered
  storage_account_name = azurerm_storage_account.cmgt-st.name
}

resource "azurerm_storage_queue" "st-cm-validation-request-processed" {
  name                 = var.st-cm-validation-request-processed
  storage_account_name = azurerm_storage_account.cmgt-st.name
}
```

### Debug Output/Panic Output

```shell
2024-03-18T11:08:42.4415510Z ##[error]Terraform command 'plan' failed with exit code '1'.
2024-03-18T11:08:42.4541358Z ##[error][31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mexpected the subdomain type to be "queue" but got "table"[0m
[31m│[0m [0m
[31m│[0m [0m[0m  with azurerm_storage_queue.st-cm-documents-created,
[31m│[0m [0m  on juno.contractormanagement.api.tf line 116, in resource "azurerm_storage_queue" "st-cm-documents-created":
[31m│[0m [0m 116: resource "azurerm_storage_queue" "st-cm-documents-created" [4m{[0m[0m
[31m│[0m [0m
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mexpected the subdomain type to be "queue" but got "table"[0m
[31m│[0m [0m
[31m│[0m [0m[0m  with azurerm_storage_queue.st-cm-connection-accepted,
[31m│[0m [0m  on juno.contractormanagement.api.tf line 121, in resource "azurerm_storage_queue" "st-cm-connection-accepted":
[31m│[0m [0m 121: resource "azurerm_storage_queue" "st-cm-connection-accepted" [4m{[0m[0m
[31m│[0m [0m
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mexpected the subdomain type to be "queue" but got "table"[0m
[31m│[0m [0m
[31m│[0m [0m[0m  with azurerm_storage_queue.st-cm-connection-rejected,
[31m│[0m [0m  on juno.contractormanagement.api.tf line 126, in resource "azurerm_storage_queue" "st-cm-connection-rejected":
[31m│[0m [0m 126: resource "azurerm_storage_queue" "st-cm-connection-rejected" [4m{[0m[0m
[31m│[0m [0m
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mexpected the subdomain type to be "queue" but got "table"[0m
[31m│[0m [0m
[31m│[0m [0m[0m  with azurerm_storage_queue.st-cm-user-updated,
[31m│[0m [0m  on juno.contractormanagement.api.tf line 131, in resource "azurerm_storage_queue" "st-cm-user-updated":
[31m│[0m [0m 131: resource "azurerm_storage_queue" "st-cm-user-updated" [4m{[0m[0m
[31m│[0m [0m
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mexpected the subdomain type to be "queue" but got "table"[0m
[31m│[0m [0m
[31m│[0m [0m[0m  with azurerm_storage_queue.st-cm-scheduled-task-triggered,
[31m│[0m [0m  on juno.contractormanagement.api.tf line 136, in resource "azurerm_storage_queue" "st-cm-scheduled-task-triggered":
[31m│[0m [0m 136: resource "azurerm_storage_queue" "st-cm-scheduled-task-triggered" [4m{[0m[0m
[31m│[0m [0m
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mexpected the subdomain type to be "queue" but got "table"[0m
[31m│[0m [0m
[31m│[0m [0m[0m  with azurerm_storage_queue.st-cm-validation-request-processed,
[31m│[0m [0m  on juno.contractormanagement.api.tf line 141, in resource "azurerm_storage_queue" "st-cm-validation-request-processed":
[31m│[0m [0m 141: resource "azurerm_storage_queue" "st-cm-validation-request-processed" [4m{[0m[0m
[31m│[0m [0m
[31m╵[0m[0m
```

**Comment by ramsinghtmdc** [comment]
Same issue i am facing while creating.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/157681942/5a4dc659-3778-467a-8f6a-7ad045c70a35)

**Comment by oscaralvreyes** [comment]
@ramsinghtmdc I fixed the issue forcing the use of the version 3.95.0

**Comment by tombuildsstuff** [comment]
hey @oscaralvreyes 

Thanks for opening this issue.

Taking a look through here this appears to be a duplicate of https://github.com/hashicorp/terraform-provider-azurerm/issues/25261 which was fixed by https://github.com/hashicorp/terraform-provider-azurerm/pull/25262. Since the fix for this issue will ship in the next release, I'm going to close this issue for the moment but I've assigned this to the next milestone, so once that's available a message will be posted here - but in the interim it should be possible to workaround this by pinning to `v3.95.0` of the AzureRM Provider.

Thanks!
